utils/fixed_decimal/src/decimal.rs

```diff
@@ -1054,6 +1068,89 @@ impl FixedDecimal {
         Self::from_str(formatted)
     }
 
+    /// Rounds the number according to the power (`n`) of 10 position.
+    ///
+    /// NOTE: This may end up adding a digit to the left!
+    ///
+    /// NOTE: This will not change the number of significant digits, it simply exists
+    /// to *round* them (and will typically reduce the size of `self.digits`)
+    ///
+    /// This function is responsible for fixing `digits`, `magnitude`, and `upper_magnitude`.
+    /// It will only modify upper_magnitude when it is not large enough to fit the rounded number.
+    /// The caller may fix up `lower_magnitude` by whatever scheme it desires
+    pub fn round(&mut self, n: u16, mode: RoundingMode) -> Result<(), Error> {
```

Comment:
@sffc , I have added this function that contains small functions for each rounding mode, SG?

Reply:
Yes, having a single function is fine.

The `n` argument should likely be an `i16`, not `u16`, since we can round both before and after the decimal separator.